modules/network/src/main/java/org/apache/ignite/network/scalecube/ScaleCubeDirectMarshallerTransport.java

```diff
@@ -0,0 +1,243 @@
+    private final DirectProcessor<Message> subject = DirectProcessor.create();
+
+    /** Message sink. */
+    private final FluxSink<Message> sink = subject.sink();
+
+    /** Close handler */
+    private final MonoProcessor<Void> stop = MonoProcessor.create();
+
+    /** On stop. */
+    private final MonoProcessor<Void> onStop = MonoProcessor.create();
+
+    /** Connection manager. */
+    private final ConnectionManager connectionManager;
+
+    /** Node address. */
+    private final Address address;
+
+    /**
+     * Constructor.
+     *
+     * @param connectionManager Connection manager.
+     */
+    public ScaleCubeDirectMarshallerTransport(ConnectionManager connectionManager) {
+        this.connectionManager = connectionManager;
+        this.connectionManager.addListener(this::onMessage);
+        this.address = prepareAddress(connectionManager.getLocalAddress());
+        // Setup cleanup
+        stop.then(doStop())
+            .doFinally(s -> onStop.onComplete())
+            .subscribe(
+                null,
+                ex -> LOG.warn("[{0}][doStop] Exception occurred: {1}", address, ex.toString())
+            );
+    }
+
+    /**
+     * Convert {@link InetSocketAddress} to {@link Address}.
+     *
+     * @param addr Address.
+     * @return ScaleCube address.
+     */
+    private static Address prepareAddress(InetSocketAddress addr) {
+        InetAddress address = addr.getAddress();
+
+        int port = addr.getPort();
+
+        if (address.isAnyLocalAddress())
+            return Address.create(Address.getLocalIpAddress().getHostAddress(), port);
+        else
+            return Address.create(address.getHostAddress(), port);
+    }
+
+    /**
+     * Cleanup resources on stop.
+     *
+     * @return A mono, that resolves when the stop operation is finished.
+     */
+    private Mono<Void> doStop() {
+        return Mono.defer(() -> {
+            LOG.info("[{0}][doStop] Stopping", address);
+
+            // Complete incoming messages observable
+            sink.complete();
+
+            LOG.info("[{0}][doStop] Stopped", address);
+            return Mono.empty();
+        });
+    }
+
+    /** {@inheritDoc} */
+    @Override public Address address() {
+        return address;
+    }
+
+    /** {@inheritDoc} */
+    @Override public Mono<Transport> start() {
+        return Mono.just(this);
+    }
+
+    /** {@inheritDoc} */
+    @Override public Mono<Void> stop() {
+        return doStop();
+    }
+
+    /** {@inheritDoc} */
+    @Override public boolean isStopped() {
+        return onStop.isDisposed();
+    }
+
+    /** {@inheritDoc} */
+    @Override public Mono<Void> send(Address address, Message message) {
+        var addr = InetSocketAddress.createUnresolved(address.host(), address.port());
+        return Mono.defer(() -> Mono.fromFuture(connectionManager.channel(addr)))
+            .flatMap(client -> {
+                client.send(fromMessage(message));
+                return Mono.empty();
```

Comment:
So we don't wait for acks here, but why?

Reply:
We don't actually have any acks, so there is no way to tell if message was received by the remote server. We can, though, implement default message handlers, that sends acks in response to any message

Reply:
Can this break membership protocol then?

Reply:
Not sure why it should. ScaleCube's default transport also doesn't have any acks.